agents-and-function-calling/bedrock-agents/features-examples/11-get-user-confirmation-before-action-execution/11.1-create-agent-with-function-definition-and-user-confirmation/11.1-create-agent-with-function-definition-and-user-confirmation.ipynb

```diff
@@ -0,0 +1,2222 @@
+      " 'sessionId': 'b4f50724-7c4e-11ef-9888-0affee3e2ef7'}\n"
+     ]
+    }
+   ],
+   "source": [
+    "agentResponse = bedrock_agent_runtime_client.invoke_agent(\n",
+    "    inputText=\"please reserve one day of time off, 2024 June 1\",\n",
+    "    agentId=agent_id,\n",
+    "    agentAliasId=agent_alias_id, \n",
+    "    sessionId=session_id,\n",
+    "    enableTrace=enable_trace, \n",
+    "    endSession= end_session\n",
+    ")\n",
+    "\n",
+    "logger.info(pprint.pprint(agentResponse))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cf4f1a53",
+   "metadata": {},
+   "source": [
+    "Different from the first example, a `returnControl` event is returned, where the `actionInvocationType` is `USER_CONFIRMATION`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 219,
+   "id": "19ac498d",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'returnControl': {'invocationId': '97c44cc1-6ed2-4918-ba88-be95e44277d0-uc',\n",
+      "                   'invocationInputs': [{'functionInvocationInput': {'actionGroup': 'VacationsActionGroup',\n",
+      "                                                                     'actionInvocationType': 'USER_CONFIRMATION',\n",
+      "                                                                     'function': 'reserve_vacation_time',\n",
+      "                                                                     'parameters': [{'name': 'end_date',\n",
+      "                                                                                     'type': 'string',\n",
+      "                                                                                     'value': '2024-06-01'},\n",
+      "                                                                                    {'name': 'employee_id',\n",
+      "                                                                                     'type': 'integer',\n",
+      "                                                                                     'value': '1'},\n",
+      "                                                                                    {'name': 'start_date',\n",
+      "                                                                                     'type': 'string',\n",
+      "                                                                                     'value': '2024-06-01'}]}}]}}\n",
+      "CPU times: user 4.6 ms, sys: 6.65 ms, total: 11.3 ms\n",
+      "Wall time: 1.26 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "event_stream = agentResponse['completion']\n",
+    "\n",
+    "for event in event_stream:\n",
+    "    if 'returnControl' in event:\n",
+    "        pprint.pp(event)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "703cba29",
+   "metadata": {},
+   "source": [
+    "Let's create a method to handle the `returnControl` event and call our lambda locally"
```

Comment:
why? this part doesn't make sense.. we should confirm the execution so that the lambda can handle it